tests/linearGaussian_vector_field_test.py

```diff
@@ -138,6 +139,84 @@ def test_c2st_vector_field_on_linearGaussian(
         )
 
 
+# We always test num_dim and sample_with with defaults and mark the rests as slow.
+# TODO: This should be better tuned to find the right balance between required precision
+# and speed, maybe avoiding to pass a fixed condition mask to the Simformer
+# but letting it generate Bernoulli ones
+@pytest.mark.parametrize(
+    "num_dim, prior_str, sample_with",
+    [
+        (3, "uniform", ["sde", "ode"]),
+        (3, "gaussian", ["sde", "ode"]),
+    ],
+)
+def test_c2st_simformer_on_linearGaussian(
+    num_dim: int, prior_str: str, sample_with: List[str]
+):
+    """
+    Test whether Simformer infers well a simple example with available ground truth.
+    """
+    # num_sim_nodes = 2  # theta, x
+    num_samples = 3000
+    num_simulations = 10000
+
+    x_o = zeros(1, num_dim)
+
+    # likelihood_mean will be likelihood_shift+theta
+    likelihood_shift = -1.0 * ones(num_dim)
+    likelihood_cov = 0.3 * eye(num_dim)
+
+    if prior_str == "gaussian":
+        prior_mean = zeros(num_dim)
+        prior_cov = eye(num_dim)
+        prior = MultivariateNormal(loc=prior_mean, covariance_matrix=prior_cov)
+        gt_posterior = true_posterior_linear_gaussian_mvn_prior(
+            x_o, likelihood_shift, likelihood_cov, prior_mean, prior_cov
+        )
+        target_samples = gt_posterior.sample((num_samples,))
+    else:
+        prior = utils.BoxUniform(-2.0 * ones(num_dim), 2.0 * ones(num_dim))
+        target_samples = samples_true_posterior_linear_gaussian_uniform_prior(
+            x_o,
+            likelihood_shift,
+            likelihood_cov,
+            prior=prior,
+            num_samples=num_samples,
+        )
+
+    # Prepare data for Simformer
+    thetas = prior.sample((num_simulations,))
+    xs = linear_gaussian(thetas, likelihood_shift, likelihood_cov)
+    # inputs shape: (num_simulations, num_nodes, num_features)
+    inputs = torch.stack([thetas, xs], dim=1)
+
+    # Infer theta (node 0) given x (node 1).
+    inference = Simformer(
+        prior=prior, show_progress_bars=True, latent_idx=[0], observed_idx=[1]
+    )
+
+    training_condition_masks = torch.tensor([False, True]).repeat(num_simulations, 1)
```

Comment:
For the general training we can easily test the posterior and e.g. the likelihood.

For the likelihood you can just sample a random theta from the prior, run multiple times the simulator on this same paramter. These samples you can then compare with what the Simformer produces for the likelihood mask.